src/Umbraco.Web.Common/Extensions/HttpRequestExtensions.cs

```diff
@@ -53,5 +56,27 @@ private static bool IsSet(this IPAddress address)
             const string NullIpAddress = "::1";
             return address != null && address.ToString() != NullIpAddress;
         }
+
+        public static string GetRawBodyString(this HttpRequest request, Encoding encoding = null)
+        {
+            request.Body.Seek(0, SeekOrigin.Begin);
+
+            var reader = new StreamReader(request.Body, encoding ?? Encoding.UTF8);
```

Comment:
Do these readers need to be disposed? I assume we don't want to dispose the stream?

Reply:
If we dispose the StreamReader it disposed the stream.. And no, we don't want the stream to be disposed

Reply:
Could we not use the overload below to dispose of the StreamReader while leaving the stream open?

https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader.-ctor?view=netcore-3.1#System_IO_StreamReader__ctor_System_IO_Stream_System_Text_Encoding_System_Boolean_System_Int32_System_Boolean_

Reply:
Nice.. Guess that will work :)